examples/wallet/app/actions/account.js

```diff
@@ -42,16 +45,65 @@ export function setAccount(privateKey: string): Thunk<SetKeysAction> {
   };
 }
 
-const initializeKeysAndRedirect = (dispatch, keys: AccountKeys) => {
+export function setAccountFromPrivateKey(
+  privateKey: string
+): Thunk<SetKeysAction> {
+  return function setAccountFromPrivateKeyThunk(dispatch) {
+    return getAccountFromPrivateKey(privateKey)
+      .then(loadedPrivateKey => {
+        dispatch({
+          type: SET_KEYS,
+          ...loadedPrivateKey
+        });
+        return Promise.all([
+          dispatch(updateAccountTransactions()),
+          dispatch(updateNodeSettings()),
+          dispatch(updateAccountState())
+        ])
+          .then(() => dispatch(push(routes.WALLET)))
```

Comment:
this error handling does not make sense to me :upside_down_face: 

You'd want to redirect to the input keys screen if there's an error with getting the AccountKeys object from the provided privatekey.
For that, I think the best would be to handle the error outside of this action (perhaps in the router action?)

You'd want to ignore the errors when fetching the balance/transactions/nodesettings, log them and redirect to the main screen anyways.

Reply:
corrected